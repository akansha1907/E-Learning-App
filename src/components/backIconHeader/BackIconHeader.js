import {View, Text, Image, Pressable} from 'react-native';
import React from 'react';
import {styles} from './styles';

const BackIconHeader = props => {
  const {title, navigation} = props || {};
  return (
    <View style={styles.container}>
      <Pressable
        onPress={() => navigation.goBack()}
        style={styles.iconContainer}>
        <Image
          source={require('../../assets/images/back_icon.png')}
          style={styles.image}
        />
      </Pressable>
      <Text style={styles.title}>{title}</Text>
    </View>
  );
};

export default BackIconHeader;
