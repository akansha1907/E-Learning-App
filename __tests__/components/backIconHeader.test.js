import React from 'react';
import BackIconHeader from '../../src/components/backIconHeader/BackIconHeader';
import {render} from '@testing-library/react-native';
describe('BackIconHeader Snapshot testing', () => {
  it('Should render BackIconHeder correctly', () => {
    const tree = render(<BackIconHeader />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
