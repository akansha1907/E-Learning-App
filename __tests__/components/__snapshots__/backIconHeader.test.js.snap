// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackIconHeader Snapshot testing Should render BackIconHeder correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignSelf": "flex-start",
        "padding": 12,
      }
    }
  >
    <Image
      source={
        {
          "testUri": "../../../src/assets/images/back_icon.png",
        }
      }
      style={
        {
          "height": 30,
          "tintColor": "#000000",
          "width": 30,
        }
      }
    />
  </View>
  <Text
    style={
      {
        "color": "#000000",
        "fontFamily": "Poppins-SemiBold",
        "fontSize": 17,
        "marginTop": 2,
        "padding": 12,
      }
    }
  />
</View>
`;
